src/category/monad/basic.lean

```diff
@@ -1,9 +1,32 @@
+/-
+Copyright (c) 2019 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Simon Hudon
+-/
 
 import tactic.basic
 
+/-!
+Set of rewrite rules and automation for monads in general and
+`reader_t`, `state_t`, `except_t` and `option_t` in particular.
+
+The rewrite rules for monads are carefully chosen so that `simp with
```

Comment:
This is useful, and the kind of information I'd like to see in all doc strings. (We've put it under the "implementation details" header in some other files.)